File: mobile/lib/models/project.dart
/// Project model — mirrors the server's Project type.
class Project {
  final String id;
  final String path;
  final ProjectConfig config;
  final ProjectStats stats;

  const Project({
    required this.id,
    required this.path,
    required this.config,
    required this.stats,
  });

  factory Project.fromJson(Map<String, dynamic> json) {
    return Project(
      id: json['id'] as String? ?? '',
      path: json['path'] as String? ?? '',
      config: ProjectConfig.fromJson(json['config'] as Map<String, dynamic>? ?? {}),
      stats: ProjectStats.fromJson(json['stats'] as Map<String, dynamic>? ?? {}),
    );
  }
}

class ProjectConfig {
  final String name;
  final String codename;
  final String? description;
  final String lifecycle;
  final String? github;
  final List<String> tags;
  final ProjectBranding? branding;

  const ProjectConfig({
    required this.name,
    required this.codename,
    this.description,
    required this.lifecycle,
    this.github,
    this.tags = const [],
    this.branding,
  });

  factory ProjectConfig.fromJson(Map<String, dynamic> json) {
    return ProjectConfig(
      name: json['name'] as String? ?? '',
      codename: json['codename'] as String? ?? '',
      description: json['description'] as String?,
      lifecycle: json['lifecycle'] as String? ?? 'planning',
      github: json['github'] as String?,
      tags: (json['tags'] as List<dynamic>?)?.cast<String>() ?? [],
      branding: json['branding'] != null
          ? ProjectBranding.fromJson(json['branding'] as Map<String, dynamic>)
          : null,
    );
  }
}

class ProjectBranding {
  final String? primaryColor;
  final String? secondaryColor;
  final String? logo;
  final String? mode;

  const ProjectBranding({this.primaryColor, this.secondaryColor, this.logo, this.mode});

  factory ProjectBranding.fromJson(Map<String, dynamic> json) {
    return ProjectBranding(
      primaryColor: json['primaryColor'] as String?,
      secondaryColor: json['secondaryColor'] as String?,
      logo: json['logo'] as String?,
      mode: json['mode'] as String?,
    );
  }
}

class ProjectStats {
  final int bc;
  final int sol;
  final int us;
  final int cmp;
  final int fn;
  final int inf;
  final int adr;
  final int ntf;
  final int conv;
  final int fbk;

  const ProjectStats({
    this.bc = 0, this.sol = 0, this.us = 0, this.cmp = 0,
    this.fn = 0, this.inf = 0, this.adr = 0, this.ntf = 0,
    this.conv = 0, this.fbk = 0,
  });

  int get total => bc + sol + us + cmp + fn + inf + adr + ntf + conv + fbk;

  factory ProjectStats.fromJson(Map<String, dynamic> json) {
    return ProjectStats(
      bc: json['bc'] as int? ?? 0,
      sol: json['sol'] as int? ?? 0,
      us: json['us'] as int? ?? 0,
      cmp: json['cmp'] as int? ?? 0,
      fn: json['fn'] as int? ?? 0,
      inf: json['inf'] as int? ?? 0,
      adr: json['adr'] as int? ?? 0,
      ntf: json['ntf'] as int? ?? 0,
      conv: json['conv'] as int? ?? 0,
      fbk: json['fbk'] as int? ?? 0,
    );
  }
}

File: _ARQITEKT/server/src/types/project.ts
/** Metamodel entity types */
export type EntityType =
  | 'BusinessCase'
  | 'Solution'
  | 'UserStory'
  | 'Component'
  | 'Function'
  | 'ConversationFlow'
  | 'Infrastructure'
  | 'ArchitectureDecision'
  | 'Notification'
  | 'Feedback';

export type EntityPrefix = 'BC' | 'SOL' | 'US' | 'CMP' | 'FN' | 'CONV' | 'INF' | 'ADR' | 'NTF' | 'FBK';

export type RequirementStatus = 'idea' | 'draft' | 'review' | 'approved' | 'implemented';

export type LifecycleStage = 'planning' | 'ready' | 'building' | 'built' | 'running' | 'deployed';

export interface ProjectConfig {
  name: string;
  codename: string;
  description?: string;
  lifecycle: LifecycleStage;
  github?: string;
  tags?: string[];
  branding?: {
    primaryColor?: string;
    secondaryColor?: string;
    logo?: string;
    mode?: string;
  };
}

export interface Project {
  id: string;
  path: string;
  config: ProjectConfig;
  stats: ProjectStats;
  readiness: ProjectReadiness;
}

export interface ProjectStats {
  bc: number;
  sol: number;
  us: number;
  cmp: number;
  fn: number;
  inf: number;
  adr: number;
  ntf: number;
  conv: number;
  fbk: number;
}

export interface ProjectReadiness {
  authored: number;
  approved: number;
}

export interface TreeNode {
  id: string;
  type: EntityPrefix;
  title: string;
  status: RequirementStatus;
  children: TreeNode[];
  parent?: string;
}

export interface RequirementFrontmatter {
  type: EntityType;
  id: string;
  title: string;
  status: RequirementStatus;
  parent?: string;
  relationships?: {
    depends_on?: string[];
    constrains?: string[];
  };
}

export interface ValidationResult {
  rule: string;
  ruleId: string;
  scope: string;
  passed: boolean;
  details?: string;
  affectedArtifacts?: string[];
}

export interface FeedbackItem {
  id: string;
  title: string;
  description: string;
  source: 'manual' | 'google-play' | 'app-store' | 'in-app' | 'email';
  severity: 'wish' | 'improvement' | 'bug' | 'critical';
  status: 'open' | 'planned' | 'done';
  rating?: number;
  createdAt: string;
}

export interface ChatMessage {
  role: 'user' | 'assistant' | 'system';
  content: string;
}

export interface AppStatus {
  running: boolean;
  port?: number;
  pid?: number;
  mode?: 'dev' | 'production';
}

/* ------------------------------------------------------------------ */
/*  Projects registry (projects.yaml)                                  */
/* ------------------------------------------------------------------ */

export type ProjectMode = 'local' | 'external';

export interface ProjectRegistryEntry {
  id: string;
  name: string;
  codename: string;
  mode: ProjectMode;
  path: string;
  github?: string;
  description?: string;
}

export interface ProjectsRegistry {
  projects: ProjectRegistryEntry[];
}

/* ------------------------------------------------------------------ */
/*  Pipeline & Gate Engine Types                                       */
/* ------------------------------------------------------------------ */

export type GateId = 'G0_IDEA_TO_BC' | 'G1_BC_TO_SOL' | 'G2_SOL_TO_US' | 'G3_US_TO_CMP' | 'G4_CMP_TO_FN' | 'G5_FN_TO_CODE';

export type GateStatus = 'passed' | 'failed' | 'pending' | 'overridden';

export type AgentType = 'socratic' | 'devils_advocate' | 'constraint' | 'example' | 'boundary';

export type RiskLevel = 'critical' | 'high' | 'medium' | 'low';

export interface GateCheck {
  id: string;
  rule: string;
  passed: boolean;
  details?: string;
  severity: RiskLevel;
}

export interface Gap {
  id: string;
  artifactId: string;
  description: string;
  severity: RiskLevel;
  suggestedAgent: AgentType;
  resolved: boolean;
}

export interface GateResult {
  gateId: GateId;
  name: string;
  from: EntityPrefix;
  to: EntityPrefix | 'CODE';
  status: GateStatus;
  confidence: number;
  checks: GateCheck[];
  gaps: Gap[];
  timestamp: string;
  overrideReason?: string;
}

export interface PipelineStatus {
  projectId: string;
  gates: GateResult[];
  overallConfidence: number;
  lastEvaluated: string;
}

export interface ConfidenceScore {
  artifactId: string;
  overall: number;
  structural: number;
  semantic: number;
  consistency: number;
  boundary: number;
  lastEvaluated: string;
}

export interface ConfidenceDetails {
  structural: number;
  semantic: number;
  consistency: number;
  boundary: number;
}

export interface GateDefinition {
  id: GateId;
  name: string;
  from: EntityPrefix;
  to: EntityPrefix | 'CODE';
  mandatoryChecks: string[];
  agents: AgentType[];
  autoPassThreshold: number;
  riskLevel: RiskLevel;
}

/* ------------------------------------------------------------------ */
/*  Probing / Agent System Types                                       */
/* ------------------------------------------------------------------ */

export interface ProbingOption {
  id: string;
  label: string;
  impact: string;
}

export interface ProbingQuestion {
  id: string;
  agentType: AgentType;
  gapId: string;
  artifactId: string;
  question: string;
  options: ProbingOption[];
  whyImportant: string;
  estimatedImpact: RiskLevel;
  canSkip: boolean;
  status: 'open' | 'answered' | 'skipped';
  answer?: string;
  skipReason?: string;
  timestamp: string;
}

export interface ProbingSession {
  projectId: string;
  artifactId: string;
  questions: ProbingQuestion[];
  startedAt: string;
  completedAt?: string;
}

export interface AgentPersona {
  type: AgentType;
  name: string;
  description: string;
  systemPrompt: string;
  triggerPatterns: string[];
  gates: GateId[];
}

/* ------------------------------------------------------------------ */
/*  Baseline & Drift Types                                             */
/* ------------------------------------------------------------------ */

export interface BaselineArtifact {
  id: string;
  type: EntityPrefix;
  title: string;
  status: RequirementStatus;
  parent?: string;
  contentHash: string;
  children: string[];
}

export interface Baseline {
  projectId: string;
  createdAt: string;
  artifacts: BaselineArtifact[];
  treeHash: string;
}

export type DriftKind =
  | 'added'
  | 'removed'
  | 'title_changed'
  | 'status_regressed'
  | 'content_changed'
  | 'parent_changed';

export interface DriftItem {
  artifactId: string;
  kind: DriftKind;
  details: string;
  severity: RiskLevel;
}

export interface DriftReport {
  projectId: string;
  baselineDate: string;
  checkedAt: string;
  drifts: DriftItem[];
  totalArtifacts: number;
  changedArtifacts: number;
}

/* ------------------------------------------------------------------ */
/*  Traceability Matrix Types                                          */
/* ------------------------------------------------------------------ */

export interface TraceLink {
  from: string;
  to: string;
  relation: 'parent' | 'child' | 'depends_on' | 'constrains';
}

export interface TraceabilityMatrix {
  projectId: string;
  links: TraceLink[];
  orphans: string[];
  leaves: string[];
}

export interface ImpactAnalysis {
  artifactId: string;
  directlyAffected: string[];
  transitivelyAffected: string[];
  totalImpact: number;
}
